cds_ils/importer/api.py

```diff
@@ -43,61 +42,68 @@ def process_dump(data, provider, mode, source_type):
 
 def import_record(data, provider, mode, source_type=None, eager=False):
     """Import record from dump."""
+    if eager:
+        return create_json_and_import(data, provider, mode,
+                                      source_type=source_type)
+    else:
+        create_json_and_import.delay(data, provider, mode,
+                                     source_type=source_type)
+
+
+def validate_import(provider, mode, source_type):
+    """Validate import."""
     source_type = source_type or "marcxml"
     assert source_type in ["marcxml"]
 
+    # check if the record is in delete mode
     if provider not in current_app.config[
         "CDS_ILS_IMPORTER_PROVIDERS_ALLOWED_TO_DELETE_RECORDS"
-    ] and mode == 'delete':
+    ] and mode == 'DELETE':
```

Comment:
```suggestion
    ] and mode == ImporterMode.DELETE.value:
```
?
To be applied everywhere?